### PR TITLE
Add support for diskqueue

### PR DESCRIPTION
**Pull request by leehinman**
Adds ability to have diskqueue in shipper.

- Relates elastic/beats#32258

This should **NOT** be merged until the changes in `server.go` are properly addressed.  The changes in this PR were just to get tests to pass.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -73,3 +74,5 @@ replace (
 	github.com/dop251/goja => github.com/andrewkroh/goja v0.0.0-20190128172624-dd2ac4456e20
 	github.com/dop251/goja_nodejs => github.com/dop251/goja_nodejs v0.0.0-20171011081505-adff31b136e6
 )
+
+replace github.com/elastic/beats/v7 v7.0.0-alpha2.0.20220627193119-7b5ef7832848 => /Users/leehinman/src/beats
```
Comment (leehinman) [inline go.mod:78]: This is necessary to use PR elastic/beats#32258, will remove this & take this PR out of draft once 32258 is merged

File: queue/config.go [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+	"fmt"
+	"time"
+
+	diskqueue "github.com/elastic/beats/v7/libbeat/publisher/queue/diskqueue"
+	memqueue "github.com/elastic/beats/v7/libbeat/publisher/queue/memqueue"
+)
+
+type Config struct {
+	Type         string             `config:"type"`
+	MemSettings  memqueue.Settings  `config:"memqueue"`
```
Comment (cmacknz) [inline queue/config.go:17]: Is there a way to make these pointers so we don't need the type parameter? You can just infer it based on whether you have mem settings or queue settings?

File: queue/config.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			Events:         1024,
+			FlushMinEvents: 256,
+			FlushTimeout:   5 * time.Millisecond,
+		}, //memqueue should have a DefaultSettings()
+		DiskSettings: diskqueue.DefaultSettings(),
+	}
+}
+
+func (c *Config) Validate() error {
+	if c.Type != "memory" && c.Type != "disk" {
```
Comment (cmacknz) [inline queue/config.go:34]: Should we validate or warn when the settings don't match the type? like if MemSettings were provided when Type is "disk"?

File: queue/config.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+func DefaultConfig() Config {
+	return Config{
+		Type: "memory",
+		MemSettings: memqueue.Settings{
+			Events:         1024,
+			FlushMinEvents: 256,
+			FlushTimeout:   5 * time.Millisecond,
+		}, //memqueue should have a DefaultSettings()
```
Comment (faec) [inline queue/config.go:28]: agreed, maybe I can fit that in my current memqueue PR

**Review by faec — APPROVED** [review]

File: queue/config.go (hunk 2 above)
Comment (leehinman) [reply queue/config.go]: switched to pointers

**Pull request merged**